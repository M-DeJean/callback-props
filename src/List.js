import React from 'react';
import Card from './Card'
import './List.css';

export default function List(props) {
  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {console.log(props.cards)}
        {props.cards.map((card) =>
          <Card
            handleDelete={props.handleDelete}
            key={card.id}
            id={card.id}
            title={card.title}
            content={card.content}
            listKey={props.id}
          />
        )}
        <button
          onClick={() => props.addCard(props.id)}
          className='List-add-button'
        > 
          + Add Random Card
        </button>
      </div>
    </section>
  )
};


