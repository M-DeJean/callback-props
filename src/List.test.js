import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import render from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List
        header = "Fourth List"
        cards = {[{id: 1, title: 'malik-tim', content: 'alex'}]}
    />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders as expected', () => {
      const tree = render.create(<List
        header = "Fourth List"
        cards = {[{id: 1, title: 'malik-tim', content: 'alex'}]}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  });
  