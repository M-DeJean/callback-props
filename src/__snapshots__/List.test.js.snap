// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Fourth List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        malik-tim
      </h3>
      <p>
        alex
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
